text/0000-proc-macros.md

```diff
@@ -0,0 +1,417 @@
+- Feature Name: procedural_macros
+- Start Date: 2016-02-15
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an evolution of Rust's procedural macro system (aka syntax
+extensions, aka compiler plugins). This RFC specifies syntax for the definition
+of procedural macros, a high-level view of their implementation in the compiler,
+and outlines how they interact with the compilation process.
+
+This RFC specifies the architecture of the procedural macro system. It relies on
+[RFC 1561](https://github.com/rust-lang/rfcs/pull/1561) which specifies the
+naming and modularisation of macros. It leaves many of the details for further
+RFCs, in particular the details of the APIs available to macro authors
+(tentatively called `libmacro`). See this [blog post](http://ncameron.org/blog/libmacro/)
```

Comment:
Yeah, that is true. We could decide to make `macro` not a keyword (is that backward compatible?). Or we could make it a contextual keyword. Or we could use a different name. I'll leave it for the libmacro rfc...